packages/waku/src/router/define-router.ts

```diff
@@ -91,15 +92,37 @@ export function unstable_defineRouter(fns: {
     elements: Record<SlotId, ReactNode>;
     fallbackElement?: ReactNode;
   }>;
+  getApiConfig?: () => Promise<
+    Iterable<{
+      path: PathSpec;
+      isStatic?: boolean;
+    }>
+  >;
+  handleApi?: (
+    path: string,
+    options: {
+      body: ReadableStream | null;
+      headers: Readonly<Record<string, string>>;
+      method: string;
+    },
+  ) => Promise<{
+    body?: ReadableStream;
+    headers?: Record<string, string | string[]>;
+    status?: number;
+  }>;
 }) {
   const platformObject = unstable_getPlatformObject();
   type MyPathConfig = {
     pathSpec: PathSpec;
     pathname: string | undefined;
     pattern: string;
-    staticElementIds: SlotId[];
-    isStatic?: boolean | undefined;
-    specs: { noSsr?: boolean; is404: boolean };
+    specs: {
+      isStatic?: true;
+      staticElementIds?: SlotId[];
+      noSsr?: true;
+      is404?: true;
+      isApi?: true;
```

Comment:
Do we still plan to have `getApiConfig` separate from `getRouteConfig`? 

Including `isApi` here makes it seem like we'll use this type for both kinds of configs, api and route.

Or is this just temporary and getApiConfig will come later?

Reply:
It is an internal implementation detail. The define-router users won't see it. Yeah, I think separating them would be good. They have different configs and I would like to pair getFooConfig and handleFoo.